raiden-dapp/tests/unit/components/channel-list.spec.ts

```diff
@@ -119,181 +107,134 @@ describe('ChannelList.vue', () => {
 
+    test('display channel deposit dialog when deposit is pressed', async () => {
+      wrapper.find('#channel-278').trigger('click');
+      await wrapper.vm.$nextTick;
+      wrapper.find('#deposit-0').trigger('click');
+
+      // @ts-ignore
+      expect(wrapper.vm.depositing).toBeTruthy();
+    });
+
+    test('deposit is called successfullt', async () => {
       // @ts-ignore
       await wrapper.vm.deposit('0.5');
-      await flushPromises();
       jest.advanceTimersByTime(2000);
 
       expect(depositInProgress).toHaveBeenNthCalledWith(1, true);
       expect(raiden.deposit).toHaveBeenCalledTimes(1);
       expect(raiden.deposit).toHaveBeenCalledWith(
-        TestData.openChannel.token,
-        TestData.openChannel.partner,
+        TestData.mockChannelArray[0].token,
+        TestData.mockChannelArray[0].partner,
         '0.5'
       );
-      expect(wrapper.emitted().message[0]).toEqual([
+      expect(wrapper.emitted().message[0][0]).toEqual(
         'channel-list.messages.deposit.success'
-      ]);
+      );
     });
 
     test('emit fail message when deposit fails', async () => {
       raiden.deposit.mockRejectedValue(new Error());
 
       // @ts-ignore
       await wrapper.vm.deposit('0.5');
-      await flushPromises();
       jest.advanceTimersByTime(2000);
 
       expect(depositInProgress).toHaveBeenNthCalledWith(1, true);
       expect(raiden.deposit).toHaveBeenCalledTimes(1);
-      expect(wrapper.emitted().message[0]).toEqual([
+      expect(wrapper.emitted().message[0][0]).toEqual(
         'channel-list.messages.deposit.failure'
-      ]);
+      );
     });
   });
 
   describe('settle a channel', () => {
-    test('settle the channel when the user confirms the action', async () => {
-      raiden.settleChannel = jest.fn().mockReturnValue('thxhash');
-      const $data = wrapper.vm.$data;
+    beforeEach(() => {
+      raiden.settleChannel = jest.fn();
+    });
 
+    test('display confirmation dialog when settle is pressed', async () => {
       wrapper.find('#channel-280').trigger('click');
-      await wrapper.vm.$nextTick();
+      await wrapper.vm.$nextTick;
       wrapper.find('#settle-2').trigger('click');
-      elementVisibilityChanged(0, 'channel-280-settle');
-      await flushPromises();
-      wrapper.find('#confirm-280').trigger('click');
-      elementVisibilityChanged(1);
-      await wrapper.vm.$nextTick();
-      expect($data.selectedChannel).toBeNull();
-
-      await flushPromises();
+
+      // @ts-ignore
+      expect(wrapper.vm.settling).toBeTruthy();
+    });
+
+    test('settle is called successfully', async () => {
+      // @ts-ignore
+      await wrapper.vm.settle();
+      jest.advanceTimersByTime(2000);
+
       expect(raiden.settleChannel).toHaveBeenCalledTimes(1);
       expect(raiden.settleChannel).toHaveBeenCalledWith(
-        TestData.settlableChannel.token,
-        TestData.settlableChannel.partner
+        TestData.mockChannelArray[0].token,
```

Comment:
Was there any specific reason for this change? Did the tests not pass when using the specific channel?

Reply:
I believe I made that change to not have to use `settlableChannel` tokens for all tests. Because it felt weird to grab the values for settlable channel tokens for deposit and close as well. Let me know if that was a faulty move from my side.

Reply:
The basic problem with this change is that while the tests pass, they are semantically wrong. The element you are using to test the settle function is an open channel. 

Before analyzing my thought, I will start with a small nitpick. I think that `TestData.settleableChannel` gives me more information about the channel compared to `mockChannelArray[0]` that is the reason I used that instead, but that is a matter of personal opinion. 

What I see as a problem in this change is that the test is semantically wrong. `mockChannelArray[0]` has an open channel that you are using to test settle. If this was the actual SDK this would not be possible. In this case, the behavior we are mocking is not the same as the one of the SDK.

I would say with a certain degree of certainty that the `settleableChannel` was not used in the deposit. The integration tests would not allow you to press deposit in a settleable channel. The open channel was used instead in these tests, to keep the semantics as close to the real SDK behavior as possible.

Also, the deposit test was using the `openChannel` but it seems that it was also replaced with `mockChannelArray[0]` from what I can see. 

![image](https://user-images.githubusercontent.com/2269732/74145148-6bc3ec80-4bfe-11ea-8ed5-dca628e13ff4.png)

Reply:
Thank you for clearifying @kelsos . That was more than I knew/considered when making the changes. :) I will adapt the tests accordingly, no problem.